src/Compilers/CSharp/Portable/FlowAnalysis/LocalDataFlowPass.cs

```diff
@@ -91,11 +91,11 @@ protected int VariableSlot(Symbol symbol, int containingSlot = 0)
         /// <summary>
         /// Force a variable to have a slot.  Returns -1 if the variable has an empty struct type.
         /// </summary>
-        protected int GetOrCreateSlot(Symbol symbol, int containingSlot = 0)
+        protected int GetOrCreateSlot(Symbol symbol, int containingSlot = 0, bool createIfMissing = true)
         {
             if (symbol.Kind == SymbolKind.RangeVariable) return -1;
 
-            containingSlot = DescendThroughTupleRestFields(ref symbol, containingSlot, forceContainingSlotsToExist: true);
+            containingSlot = DescendThroughTupleRestFields(ref symbol, containingSlot, createIfMissing);
```

Comment:
>containingSlot = DescendThroughTupleRestFields(ref symbol, containingSlot, createIfMissing); [](start = 12, length = 92)

I understand that this is an existing logic. It looks like ```DescendThroughTupleRestFields``` can return ```-1```. Does it make sense to continue in this case? It looks like we will go ahead and allocate a new slot for the member even in this case. It also looks like that slot is going to be shared across distinct original containers, effectively treating members of different containers as the same storage location.